packages/utils/src/types/primitives.ts

```diff
@@ -251,6 +251,10 @@ export class Balance {
     return `0x${this.bn.toString('hex', 2 * Balance.SIZE)}`
   }
 
+  public add(b: Balance): Balance {
+    return new Balance(this.bn.add(b.toBN()))
```

Comment:
In theory, you should add `.mod((2**256) - 1` or throw an error as done in the smart contract

Reply:
Please add a separate ticket for this.

Reply:
https://github.com/hoprnet/hoprnet/issues/1796